File: src/main/java/com/turbointernational/metadata/domain/type/PartTypeController.java
package com.turbointernational.metadata.domain.type;
import com.turbointernational.metadata.domain.type.PartType;
import org.springframework.roo.addon.web.mvc.controller.scaffold.RooWebScaffold;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

@RequestMapping("/parttypes")
@Controller
@RooWebScaffold(path = "parttypes", formBackingObject = PartType.class)
public class PartTypeController {
}

File: metadata-refactored/src/main/kotlin/com/turbointernational/metadatarefactored/metadatarefactored/controller/PartTypeController.kt
package com.turbointernational.metadatarefactored.metadatarefactored.controller

import com.turbointernational.metadatarefactored.metadatarefactored.dto.PartTypeDto
import com.turbointernational.metadatarefactored.metadatarefactored.service.PartService
import com.turbointernational.metadatarefactored.metadatarefactored.service.PartTypeService
import org.springframework.web.bind.annotation.*

/**
 * ruslan.klymenko@zorallabs.com 08.05.18
 */
@RestController
class PartTypeController(private val partTypeService: PartTypeService,
                         private val partService: PartService) {

    @PostMapping("/parttype")
    fun postPartType(@RequestBody partTypeDto: PartTypeDto) {
        partTypeService.create(partTypeDto)
    }

    @GetMapping("/parttype")
    fun getPartTypes() = partTypeService.getPartTypes()

    @GetMapping("/parttype/{id}/part")
    fun getPartsByPartTypes(@PathVariable("id") partId: Long) = partService.getPartsByTypeId(partId)
}